### DISCUSSION TITLE
Bash Hook Engine/Func

### DISCUSSION THREAD

**Discussion by JaylinYu** [nanomq/nanomq#297] [Ideas]
For anyone who sees it.
I am considering building a Hook Engine to hook bash script with MQTT event.
Basically it process your bash command asynchronously while triggering MQTT events like "online, offline, publish, Suback etc".
I believe it is a useful feature for building event-driven systems on edge. However, I am not sure it is the right move to do so.

So what says you.

**Comment by hujung-de** [comment]
the idea is fine ... 
let's see the 3 ways to work on a remote system ...
1. ssh/sftp : common usage BUT ops need an account, access rights, certs, ...
2. bash-service of a message broker : nice tool for ops without account, access right, certs on the target machine BUT the guys from the IT Security will not really like that service
3. message broker itself : nice and mostly invisible way for ops to do their job BUT the guys from the IT Security will find out this smart way to work on the machine and are not really amused 
  
From the TechySide it's not the problem to implement that functionality BUT to continue with this idea first discuss it with the guys in black suits, wearing nice ties, sitting near the management and play IT Security
